### ISSUE TITLE
Subs: connect existing Subscription and Orders with User created after them

### ISSUE THREAD

**Issue by kirstenalarsen** [openfoodfoundation/openfoodnetwork#2511]
## Description
There are a complex of issues connected to the case where a Hub adds a Customer that is not an existing confirmed OFN user, and then creates a Subscription for them. When the Customer tries to click the link to access their order they can't as they're not an OFN user. (#2506 will improve messaging and action on that link)

#2508 suggests that this case should be prevented by not allowing creation of a Subscription without a confirmed OFN User already existing. 

This Issue 'floats' a possible alternative response in which the Subscription can be created but the Customer needs to sign up and/or login to view or change their Orders.

## Expected Behavior
A Hub creates a Customer who is not already a user on OFN. The Hub then creates a Subscription for that Customer.

When the Order is generated, the email confirmation link brings up login / sign-up panel (#2506). The Customer must sign-up / login to be able to view or edit their order

When the Customer signs up they will receive an email confirmation, they need to validate their email, then a User is created.  

**_The system must then MATCH / reassign any existing 'guest' orders to that new User so that when they login / use the link in their email again they can see the orders created for their guest Customer before they had a User._**

## Actual Behavior
The Customer can create a User but pre-existing orders cannot be viewed or edited - they are lost in cyberspace. 

## Steps to Reproduce
1. Create a Customer that is not a User
2. Create a Subscription
3. Have that subscription generate an order 
4. Customer clicks on link
![image](https://user-images.githubusercontent.com/2911176/43374033-f34e2e58-93f0-11e8-8c16-0212af2ca8e6.png)
5. Customer signs-up and creates User
6. Customer completes email confirmation
7. Customer goes to 'my account' or clicks on link to access order again
8. No orders visible - orders created previously as Guest on this email are not recognised as linked to this User

## Severity
S2 - either this or #2508 needs to be resolved for Subscriptions V1 to be useable

## Possible Fix
* Make pre-existing Guest orders visible to User created using that email, including editable if OC still open and is allowed
* #2506 to prompt login / sign-up

**Comment by mkllnk** [comment]
We talked about connecting guest orders and newly signed up users before. The problem is that we can't assume that all guest orders for one email address actually belong to the same person. Email addresses can change owners like phone numbers.

There are many solutions to this problem, but the best solutions are also a lot of work. If you want a quick fix, it would be easy to not allow subscriptions for guests or don't tell them that they can edit their order (hide that text for guests). In version 1, the subscription is set up and changed via the shop manager anyway. So it would be reasonable to call or email the manager to change the order, I think.

**Comment by kirstenalarsen** [comment]
Thanks @mkllnk and BANG I reckon you hit on a workable solution there - @sstead @myriamboure what do you think of Maikel's idea - just remove the text and link about amending your order if they do not have a confirmed user?

Gets rid of this problem for now and options in #2508 can be considered in Subs next steps?

**Comment by myriamboure** [comment]
OK it seems the same conversation is happening in 3 GH issue, it's a bit hard to follow... @kirstenalarsen  I understand your will to fix ASAP to make v1 work and have no objection if you want to remove the link about amending your order for unconfirmed users. But I'm also aware that on a longer term, if the user finally creates an account, their past orders will never be associated for any of the automatic orders made before. So IMO we are creating problems for the future. I think the more we get out of the possibility to let people make orders as guest and the quicker we do it, the less problems we will have in the future. I think I would rather trigger a confirmation email as soon as a subscription is created for a user and make sure no subscription can be made for shoppers who are not users. It might not be such a big work... did you discuss that option with @oeoeaio and Tshirt sized it?

**Comment by sstead** [comment]
It feels messy (in terms of being able to tell shop keepers and customers what to expect regarding viewing/editing subs) but if others like this approach, I support it as a short term thing.

________________________________

**Comment by mkllnk** [comment]
> I would rather trigger a confirmation email as soon as a subscription is created for a user and make sure no subscription can be made for shoppers who are not users.

There are two tasks here:

1. Prevent subscriptions for guest users. I'm not familiar enough with the interface to say how hard it is, but it seems small while the email text change is extra small.
2. Create a user account automatically when a subscription is created. This is helped by the recent work in the admin interface around creating users. I would see this as medium sized, not epic. But again, I haven't worked with the subscription code yet. I don't think that there are any permission problems, because anyone can just go to the register form and trigger that email for a new account for any known email address.

**Comment by kirstenalarsen** [comment]
ok. So I think we agree that this as specified above is not the correct solution for this problem. I am:
* creating new issue @2534, which along with #2506 gives us a quick fix solution
* closing this issue
* updating the ['Subscriptions Improvements'](https://community.openfoodnetwork.org/t/subscriptions-improvements/1349/15) discourse thread so these issues can be thought through properly for next version
